tests/AuthorizationTest.php

```diff
@@ -123,9 +123,11 @@ function (Update $u): string { return 'dummy'; },
 
     public function provideApplicableCookieDomains(): iterable
     {
-        yield ['demo.example.com', 'https://demo.example.com', 'https://example.com'];
-        yield ['mercure.example.com', 'https://mercure.example.com', 'https://app.example.com'];
-        yield ['example.com', 'https://example.com/.well-known/mercure', 'https://app.example.com'];
+        yield ['.example.com', 'https://foo.bar.baz.example.com', 'https://foo.bar.baz.qux.example.com'];
+        yield ['.foo.bar.baz.example.com', 'https://mercure.foo.bar.baz.example.com', 'https://app.foo.bar.baz.example.com'];
+        yield ['example.com', 'https://demo.example.com', 'https://example.com'];
+        yield ['.example.com', 'https://mercure.example.com', 'https://app.example.com'];
+        yield ['.example.com', 'https://example.com/.well-known/mercure', 'https://app.example.com'];
```

Comment:
This one I'm not sure about.

in [RFC2109](www.ietf.org/rfc/rfc2109.txt) it says:

```text
   * A is a FQDN string and has the form NB, where N is a non-empty name
     string, B has the form .B', and B' is a FQDN string.  (So, x.y.com
     domain-matches .y.com but not y.com.)
```

cc @dunglas

Reply:
and we can't set the cookie for example.com from app.example.com 😶

Reply:
This RFC is outdated, see https://stackoverflow.com/a/23086139/1352334 and https://stackoverflow.com/a/23086139/1352334 for the current rules.

So this works.